packages/react-transform/test/node/index.test.tsx

```diff
@@ -676,4 +676,265 @@ describe("React Signals Babel Transform", () => {
+					signal.value;
+					return _jsx("div", { children: "Hello World" });
+				};
+			`;
+
+			// Should not transform because jsx-runtime detection is disabled - no useSignals import should be added
+			const expectedOutput = `
+				import { jsx as _jsx } from "react/jsx-runtime";
+				function MyComponent() {
+					signal.value;
+					return _jsx("div", {
+						children: "Hello World",
+					});
+				}
+			`;
+
+			runTest(inputCode, expectedOutput, {
+				experimental: { detectJSXAlternatives: false },
+			});
+		});
+
+		it("detects createElement calls created using react import", () => {
+			const inputCode = `
+				import { createElement } from "react";
+				function MyComponent() {
+					signal.value;
+					return createElement("div", { children: "Hello World" });
+				};
+			`;
+
+			const expectedOutput = `
+				import { createElement } from "react";
+				import { useSignals as _useSignals } from "@preact/signals-react/runtime";
+				function MyComponent() {
+					var _effect = _useSignals(1);
+					try {
+						signal.value;
+						return createElement("div", {
+							children: "Hello World",
+						});
+					} finally {
+						_effect.f();
+					}
+				}
+			`;
+
+			runTest(inputCode, expectedOutput, {
+				experimental: { detectJSXAlternatives: true },
+			});
+		});
+
+		it("detects createElement calls created using react default import", () => {
+			const inputCode = `
+				import React from "react";
+				function MyComponent() {
+					signal.value;
+					return React.createElement("div", { children: "Hello World" });
+				};
+			`;
+
+			const expectedOutput = `
+				import React from "react";
+				import { useSignals as _useSignals } from "@preact/signals-react/runtime";
+				function MyComponent() {
+					var _effect = _useSignals(1);
+					try {
+						signal.value;
+						return React.createElement("div", {
+							children: "Hello World",
+						});
+					} finally {
+						_effect.f();
+					}
+				}
+			`;
+
+			runTest(inputCode, expectedOutput, {
+				experimental: { detectJSXAlternatives: true },
+			});
+		});
+
+		it("detects createElement calls created using react cjs require", () => {
+			const inputCode = `
+				const React = require("react");
+				function MyComponent() {
+					signal.value;
+					return React.createElement("div", { children: "Hello World" });
+				};
+			`;
+
+			const expectedOutput = `
+				var _useSignals = require("@preact/signals-react/runtime").useSignals
+				const React = require("react");
+				function MyComponent() {
+					var _effect = _useSignals(1);
+					try {
+						signal.value;
+						return React.createElement("div", {
+							children: "Hello World",
+						});
+					} finally {
+						_effect.f();
+					}
+				}
+			`;
+
+			runTest(
+				inputCode,
+				expectedOutput,
+				{
+					experimental: { detectJSXAlternatives: true },
+				},
+				undefined,
+				true
+			);
+		});
+
+		it("detects createElement calls created using destructured react cjs require", () => {
```

Comment:
Can we add a named `.map` function i.e.

```
function X() {
  return <div>{array.map(function Y() {}})
}
```

Reply:
Hey, I’m not sure I understand exactly what you mean here, sorry. 😅 

Could you please elaborate / give a larger example? 

Thank you!

Reply:
Can you check whether a nested JSX function like i.e. a map inside of the return body leads to any false positives